### DISCUSSION TITLE
Regular calendar event view

### DISCUSSION THREAD

**Discussion by Akselmo** [TechbeeAT/jtxBoard#1210] [Ideas]
I'd love to have everything in one app. There's really no apps that can do both calendar and tasks in same app for android (caldav, not google calendar)

Any possibility for regular calendar view, with list of calendar events? Having everything in one app would be really nice.

**Comment by rogercreagh** [comment]
Personally I don't like this idea as it seems like feature bloat and is unnecessary. There are plenty of calendar apps out there. You have a multi-functional device (phone/tablet), why do you need multi-functional aps on it? I prefer apps that do one thing well rather than try to do everything poorly. 

For me the core essential feature of jtxB is using VJounral for journal/diary entries and notes sync'd across my devices through CalDav - there is nothing else that does this. 

Already there are many compromises in jtxB because of the inclusion of Tasks - there are plenty of other apps that handle task well and when I do need a ToDo list I'm happy to use a different app.

My preference here would be to actually see jtxB split into two separate apps - jtJournals and jtxTasks which would each be optimised for their specific area. jtxCalendar would be a third app. There could be cross links between them (which I guess from the little I know about Android might be difficult to implement) so that for example a task could cross link to a journal entry and vice versa.

I'm aware that many people seem to be using jtxB primarily for the Tasks functionality - but for me I would be delighted if all that stuff was spun off into a dedicated app and we got back to the core Journal functionality.

**Reply by Akselmo** [reply]
My problem is that i have to jump between two apps to check what happens when.. And i tend to forget to do that.

**Reply by rogercreagh** [reply]
Ah, so you are switching between Tasks and Calendar I guess. Journal entries are generally records of past events not future ones (for me at least) and Notes are Jounral entries without a date at all so that issue doesn't arise. Journals and notes are handled perfectly well within jtxB I find. 

The thing I'd like is a Nextcloud notes app which used CalDav - the current official one doesn't for historical reasons which is a pain as there is no way to see Journal or note entries online in Nextcloud. 

For tasks I find a dedicated task app is best for organising them and most calendar/email apps have the ability to at least display tasks and task reminders which suffices for seeing what is due when in with calendar appointments. I don't regard jtxB as a Tasks app and don't use it for that.

**Reply by Akselmo** [reply]
Yes, i use tasks and calendar both a lot. Unfortunately i cant find any FOSS calendar apps that can show tasks as well. 

Basically i would be fine with using another app for tasks as well as long as i could have a calendar view that shows both events and tasks. 

So thats where this idea came from: Since we already have tasks, notes and journals, why not calendar too? And of course make it a module one can disable.

But if this app wants to stay as tasks/notes/journals app only i can respect that. Just need to find a calendar app that can show tasks too, that's all.

**Reply by boxallar** [reply]
I do not wish to be disloyal, but maybe one of these can help you.
[SimpleCalendar](https://github.com/SimpleMobileTools/Simple-Calendar) - I have not tried it, but it says "A simple calendar with **events**, **tasks**, customizable colors, widgets and no ads."
Or maybe [ToDoAgenda](https://f-droid.org/en/packages/org.andstatus.todoagenda/) could work for you?
Could you simply add all your tasks to your calendar app as events? Or all your events as scheduled tasks in your notes app - Then jtxBoard, might work for you!

**Reply by Akselmo** [reply]
SimpleCalendar is okay, but the tasks system is their own thing and wont sync over nextcloud, which is what i need :/

ToDoAgenda is a widget which i do like a lot though!

> Could you simply add all your tasks to your calendar app as events

Hm, that's a possibility. I rather keep the two worlds separate though, but I'll consider it.

Thanks for the help though, I appreciate it.

**Reply by rogercreagh** [reply]
I find the calendars in both Thunderbird and Evolution email clients display tasks well enough. Not sure if either is available on Android/iOS though. 

I've just discovered that all the Simple Tools apps including SimpleCalendar have been sold to an ad company and are no longer FOSS and no longer updated. Shame as I use most of them.

there is a fork at https://github.com/FossifyOrg which is gradually being released on fDroid with links to the sources for the apps and discussions/feature requests. CalDAV for tasks has been proposed - the more people pile in to support it the more likely it is to happen.

**Reply by patrickunterwegs** [reply]
Hi, so basically from my perspective I planned to do a basic calendar integration. I don't want to do a fully fledged calendar app though. The idea was more to have some basic views (the same as now for journals) and show the calendar entries to allow users to link tasks and journals to an event in the calendar. I am thinking about doing this basic integration later this year when I quit my current job.

I understand also the desire to have a dedicated journals app. But here the constraint is more in DAVx5 which currently handles journals and tasks apps as one - so the change must be made first there to provide a selector for a journals and another for a tasks app.

About other calendar apps you could check out aCalendar+, I have been told this one integrated tasks from OpenTasks and Tasks.org.

**Reply by Akselmo** [reply]
Thanks, aCalendar+ is pretty much what i was looking for. Unfortunate that its not open, but anyway.

I will follow the progress of jtx board however :)

**Reply by rogercreagh** [reply]
> I understand also the desire to have a dedicated journals app. But here the constraint is more in DAVx5 which currently handles journals and tasks apps as one - so the change must be made first there to provide a selector for a journals and another for a tasks app.

Is it worth considering dropping DavX5 and using a CalDAV sync library built in to the app - might speed things up if it was only syncing one type of element.

**Comment by patrickunterwegs** [comment]
> Is it worth considering dropping DavX5 and using a CalDAV sync library built in to the app - might speed things up if it was only syncing one type of element.

I have been thinking about this a few times, but it's a very complex topic and I can only do this when I know that I will have a lot of time to dedicate on this topic for a few weeks. So this will still take some time until I start doing this. Until then I want to focus on smaller topics.

**Comment by boxallar** [comment]
> I find the calendars in both Thunderbird and Evolution email clients display tasks well enough. Not sure if either is available on Android/iOS though.
There is a Thunderbird Android app, they recently merged with K9-Mail I believe. I do not know how similar it is to the desktop version of Thunderbird however.

**Reply by patrickunterwegs** [reply]
As far as I know K9-Mail was taken over by Mozilla/Thunderbird, but it was a standalone E-Mail client before, I don't think that they will add major calendar or tasks functionality any time soon.
